source/distributor/service_impl.h

```diff
@@ -0,0 +1,54 @@
+#pragma once
+#ifdef __clang__
+#pragma clang diagnostic push
+#pragma clang diagnostic warning "-Wunused-parameter"
+#endif
+
+#include <tuple>
+
+#include "api/distributor/distributor.grpc.pb.h"
+
+#ifdef __clang__
+#pragma clang diagnostic pop
+#endif
+
+#include "external/envoy/source/common/common/logger.h"
+#include "external/envoy/source/common/common/statusor.h"
+
+#include "nighthawk/common/nighthawk_service_client.h"
+
+namespace Nighthawk {
+
+/**
+ * Implements a real-world distributor gRPC service.
+ */
+class NighthawkDistributorServiceImpl final
+    : public nighthawk::NighthawkDistributor::Service,
+      public Envoy::Logger::Loggable<Envoy::Logger::Id::main> {
+public:
+  /**
+   * Construct a new gRPC distributor service instance.
+   *
+   * @param service_client gRPC client that will be used to communicate with Nighthawk's load
+   * generator services.
+   */
+  NighthawkDistributorServiceImpl(std::unique_ptr<NighthawkServiceClient> service_client)
+      : service_client_(std::move(service_client)) {}
+
+  grpc::Status DistributedRequestStream(
+      grpc::ServerContext* context,
+      grpc::ServerReaderWriter<nighthawk::DistributedResponse, nighthawk::DistributedRequest>*
+          stream) override;
+
+private:
+  grpc::Status validateRequest(const nighthawk::DistributedRequest& request) const;
```

Comment:
(optional) Do all of these need to be on the class? Or can some of them just reside in the anonymous namespace in the .cc file without class access (maybe with small amounts of parameters passed in)?

Anonymous namespace being the smallest scope we can give a function requires less attention from any future readers.

Reply:
Done in 6471b8687332bcae77a637c8b1d86640299e20c6